diff -urN anjuta-2.1.1/configure.in anjuta-2.1.1.new/configure.in
--- anjuta-2.1.1/configure.in	2007-02-19 17:59:48.000000000 +0100
+++ anjuta-2.1.1.new/configure.in	2007-02-19 18:13:51.000000000 +0100
@@ -344,11 +344,6 @@
 				libgladeui_found="no"
 			])
 	fi
-	if test x$libgladeui_found = xold; then
-		libgladeui_found="yes"
-		AC_DEFINE_UNQUOTED(GLADEUI_LAST_PUBLIC, [1], [Last public GLADEUI version])
-		AC_SUBST(GLADEUI_LAST_PUBLIC)
-	fi
 	if test x$libgladeui_found = xno; then
 		PKG_CHECK_MODULES(PLUGIN_GLADE, [libgladeui-1.0 = $GLADEUI_REQUIRED],
 			[
