--- anjuta-0.1.8/global-tags/Makefile.am.old	Tue Feb 12 00:13:56 2002
+++ anjuta-0.1.8/global-tags/Makefile.am	Mon Feb 11 22:22:43 2002
@@ -14,7 +14,7 @@
 
 LDADD = -L$(top_srcdir)/tagmanager $(GLIB_LIBS) -lctags
 
-all-local: system.tags
+all-local:
 
 system.tags: create_global_tags.sh tm_global_tags
 	echo "Detecting header files and Generating global symbols ....."
diff -ur anjuta-0.1.8/data/Makefile.am anjuta-0.1.8-new/data/Makefile.am
--- anjuta-0.1.8/data/Makefile.am	Tue Feb 12 18:11:32 2002
+++ anjuta-0.1.8-new/data/Makefile.am	Tue Feb 12 18:40:51 2002
@@ -18,8 +18,7 @@
 	autogen.sh.gtk\
 	autogen.sh.generic\
 	autogen.sh.gtkmm\
-	autogen.sh.gnomemm\
-	system.tags
+	autogen.sh.gnomemm
 
 EXTRA_DIST = $(anjuta_data_DATA)
 
