diff -aurN anjuta-2.0.2.orig/configure.in anjuta-2.0.2/configure.in
--- anjuta-2.0.2.orig/configure.in	2006-05-15 21:37:47.000000000 +0200
+++ anjuta-2.0.2/configure.in	2006-08-13 13:55:01.590512093 +0200
@@ -84,8 +84,8 @@
     CXXFLAGS_WARNINGS=""
 fi
 
-AM_CFLAGS="$AM_CFLAGS -O0 -g $CFLAGS_WARNINGS"
-AM_CXXFLAGS="$AM_CXXFLAGS -O0 -g $CXXFLAGS_WARNINGS"
+AM_CFLAGS="$AM_CFLAGS $CFLAGS_WARNINGS"
+AM_CXXFLAGS="$AM_CXXFLAGS $CXXFLAGS_WARNINGS"
 
 dnl Enable debugging mode
 AC_ARG_ENABLE(debug,
