diff -urN anjuta-1.2.2.orig/anjuta.desktop.in.in anjuta-1.2.2/anjuta.desktop.in.in
--- anjuta-1.2.2.orig/anjuta.desktop.in.in	2004-11-17 12:00:01.710546816 +0100
+++ anjuta-1.2.2/anjuta.desktop.in.in	2004-11-17 12:01:03.779110960 +0100
@@ -6,6 +6,6 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=GNOME;GTK;Development;
+Categories=GTK;GNOME;Development;IDE;
 Encoding=UTF-8
 StartupNotify=true
