diff -Nur anjuta/data/Makefile.am anjuta.new/data/Makefile.am
--- anjuta/data/Makefile.am	Thu Feb 20 20:17:05 2003
+++ anjuta.new/data/Makefile.am	Sat Mar  8 10:39:27 2003
@@ -48,14 +48,14 @@
 EXTRA_DIST = $(anjuta_data_DATA) $(anjuta_mime_DATA) $(anjuta_mimelnk_DATA)
 
 install-data-local: system.tags
-	mkdir -p $(MACROS_dir)
-	mkdir -p $(MACROS2_dir)
-	$(INSTALL_DATA) $(MACROS:%=$(srcdir)/macros/%) $(MACROS_dir)
-	$(INSTALL_DATA) $(MACROS2:%=$(srcdir)/macros2/%) $(MACROS2_dir)
-	mv $(MACROS_dir)/Makefile.am.project $(MACROS_dir)/Makefile.am
-	mv $(MACROS2_dir)/Makefile.am.project $(MACROS2_dir)/Makefile.am
-	$(INSTALL_DATA) system.tags $(anjuta_datadir)
-	$(INSTALL_DATA) $(top_srcdir)/setup-gettext $(anjuta_datadir)
+	mkdir -p $(DESTDIR)$(MACROS_dir)
+	mkdir -p $(DESTDIR)$(MACROS2_dir)
+	$(INSTALL_DATA) $(MACROS:%=$(srcdir)/macros/%) $(DESTDIR)$(MACROS_dir)
+	$(INSTALL_DATA) $(MACROS2:%=$(srcdir)/macros2/%) $(DESTDIR)$(MACROS2_dir)
+	mv $(DESTDIR)$(MACROS_dir)/Makefile.am.project $(DESTDIR)$(MACROS_dir)/Makefile.am
+	mv $(DESTDIR)$(MACROS2_dir)/Makefile.am.project $(DESTDIR)$(MACROS2_dir)/Makefile.am
+	$(INSTALL_DATA) system.tags $(DESTDIR)$(anjuta_datadir)
+	$(INSTALL_DATA) $(top_srcdir)/setup-gettext $(DESTDIR)$(anjuta_datadir)
 
 uninstall-local:
 	@rm -fv $(anjuta_datadir)/system.tags
