diff -uNr anjuta-0.1.9.orig/Makefile.am anjuta-0.1.9/Makefile.am
--- anjuta-0.1.9.orig/Makefile.am	Thu Feb 14 20:28:34 2002
+++ anjuta-0.1.9/Makefile.am	Tue Sep 17 10:36:46 2002
@@ -3,7 +3,7 @@
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
 
-SUBDIRS = intl po macros tagmanager global-tags scintilla \
+SUBDIRS = intl po macros tagmanager scintilla \
 	widgets manuals omf-install src pixmaps data \
 	doc launcher plugins scripts
 
diff -uNr anjuta-0.1.9.orig/configure.in anjuta-0.1.9/configure.in
--- anjuta-0.1.9.orig/configure.in	Thu Feb 14 20:28:34 2002
+++ anjuta-0.1.9/configure.in	Tue Sep 17 10:39:02 2002
@@ -364,7 +364,6 @@
 macros/Makefile
 tagmanager/Makefile
 tagmanager/include/Makefile
-global-tags/Makefile
 scintilla/Makefile
 widgets/Makefile
 manuals/Makefile
diff -uNr anjuta-0.1.9.orig/data/Makefile.am anjuta-0.1.9/data/Makefile.am
--- anjuta-0.1.9.orig/data/Makefile.am	Tue Sep 17 11:27:53 2002
+++ anjuta-0.1.9/data/Makefile.am	Tue Sep 17 11:28:50 2002
@@ -30,7 +30,7 @@
 
 MACROS_dir = $(prefix)/@NO_PREFIX_PACKAGE_DATA_DIR@/gnome
 
-install-data-local: system.tags
+install-data-local: 
 	mkdir -p $(DESTDIR)$(MACROS_dir)
 	$(INSTALL_DATA) $(MACROS:%=$(top_srcdir)/macros/%) $(DESTDIR)$(MACROS_dir)
 	$(INSTALL_DATA) system.tags $(DESTDIR)$(anjuta_datadir)
@@ -48,9 +48,3 @@
 
 clean-local:
 	@rm -f system.tags
-
-system.tags:
-	cd $(top_builddir)/tagmanager && make libctags.a
-	cd $(top_builddir)/global-tags && make system.tags
-	mv $(top_builddir)/global-tags/system.tags .
-
diff -uNr anjuta-0.1.9.orig/data/Makefile.am anjuta-0.1.9/data/Makefile.am
--- anjuta-0.1.9.orig/data/Makefile.am	Tue Sep 17 11:55:19 2002
+++ anjuta-0.1.9/data/Makefile.am	Tue Sep 17 11:56:09 2002
@@ -33,11 +33,9 @@
 install-data-local: 
 	mkdir -p $(DESTDIR)$(MACROS_dir)
 	$(INSTALL_DATA) $(MACROS:%=$(top_srcdir)/macros/%) $(DESTDIR)$(MACROS_dir)
-	$(INSTALL_DATA) system.tags $(DESTDIR)$(anjuta_datadir)
 
 uninstall-local:
 	@rm -fv $(MACROS:%=$(MACROS_dir)/%)
-	@rm -fv $(anjuta_datadir)/system.tags
 	if test -d $(MACROS_dir); then \
 	  rmdir $(MACROS_dir); \
 	  \
@@ -47,4 +45,3 @@
 	fi
 
 clean-local:
-	@rm -f system.tags
