diff -urN aa/anjuta-1.2.2/configure.in anjuta-1.2.2/configure.in
--- aa/anjuta-1.2.2/configure.in	2004-04-12 10:10:34.000000000 +0200
+++ anjuta-1.2.2/configure.in	2004-04-14 13:20:49.848974256 +0200
@@ -178,7 +178,7 @@
 dnl ***************************************************************************
 dnl Internatinalization
 dnl ***************************************************************************
-ALL_LINGUAS="az be ca cs da de el en_CA en_GB es fr ga hr hi it ja ko mk ml ms nl no pl pt pt_BR ru sk sq sr sr@Latn sv tr uk zh_CN zh_TW"
+ALL_LINGUAS="az be ca cs da de el en_CA en_GB es fr ga hr hi it ja ko mk ml ms nb nl pl pt pt_BR ru sk sq sr sr@Latn sv tr uk zh_CN zh_TW"
 GETTEXT_PACKAGE=anjuta
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [GETTEXT package name])
