diff -urN anjuta-2.1.0/global-tags/create_global_tags.sh.in anjuta-2.1.0.new/global-tags/create_global_tags.sh.in
--- anjuta-2.1.0/global-tags/create_global_tags.sh.in	2007-01-02 09:59:19.000000000 +0100
+++ anjuta-2.1.0.new/global-tags/create_global_tags.sh.in	2007-01-18 18:34:58.000000000 +0100
@@ -6,7 +6,7 @@
     BASEDIR=`pwd`
     PROGDIR=. # `dirname $0`
     CFLAGS=""
-    ANJUTA_TAGS_BIN="@scriptdir@/anjuta-tags"
+    ANJUTA_TAGS_BIN="/usr/bin/anjuta-tags"
     OUTPUT_FILE=$1
     shift
     
