diff -uNr anjuta-0.1.9.orig/Makefile.am anjuta-0.1.9/Makefile.am
--- anjuta-0.1.9.orig/Makefile.am	Thu Feb 14 20:28:34 2002
+++ anjuta-0.1.9/Makefile.am	Tue Sep 17 10:36:46 2002
@@ -3,7 +3,7 @@
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
 
-SUBDIRS = intl po macros tagmanager global-tags scintilla \
+SUBDIRS = intl po macros tagmanager scintilla \
 	widgets printing manuals omf-install pixmaps scripts data \
 	doc launcher plugins src macros2
 
diff -uNr anjuta-0.1.9.orig/configure.in anjuta-0.1.9/configure.in
--- anjuta-0.1.9.orig/configure.in	Thu Feb 14 20:28:34 2002
+++ anjuta-0.1.9/configure.in	Tue Sep 17 10:39:02 2002
@@ -364,7 +364,6 @@
 macros/Makefile
 tagmanager/Makefile
 tagmanager/include/Makefile
-global-tags/Makefile
 scintilla/Makefile
 widgets/Makefile
 manuals/Makefile
diff -uNr anjuta-0.1.9.orig/data/Makefile.am anjuta-0.1.9/data/Makefile.am
--- anjuta-0.1.9.orig/data/Makefile.am	Tue Sep 17 11:27:53 2002
+++ anjuta-0.1.9/data/Makefile.am	Tue Sep 17 11:28:50 2002
@@ -45,17 +45,15 @@
 
 MACROS2_dir = $(prefix)/@NO_PREFIX_PACKAGE_DATA_DIR@/gnome2
 
-install-data-local: system.tags
+install-data-local:
 	mkdir -p $(DESTDIR)$(MACROS_dir)
 	mkdir -p $(DESTDIR)$(MACROS2_dir)
 	$(INSTALL_DATA) $(MACROS:%=$(top_srcdir)/macros/%) $(DESTDIR)$(MACROS_dir)
 	$(INSTALL_DATA) $(MACROS2:%=$(top_srcdir)/macros2/%) $(DESTDIR)$(MACROS2_dir)
 	mv $(DESTDIR)$(MACROS2_dir)/Makefile.am.project $(DESTDIR)$(MACROS2_dir)/Makefile.am
-	$(INSTALL_DATA) system.tags $(DESTDIR)$(anjuta_datadir)
 	$(INSTALL_DATA) $(top_srcdir)/setup-gettext $(DESTDIR)$(anjuta_datadir)
 
 uninstall-local:
-	@rm -fv $(anjuta_datadir)/system.tags
 	@rm -fv $(anjuta_datadir)/setup-gettext
 	mv $(MACROS2_dir)/Makefile.am $(MACROS2_dir)/Makefile.am.project
 	@rm -fv $(MACROS:%=$(MACROS_dir)/%)
@@ -55,9 +55,3 @@
 	    rmdir $(prefix)/@NO_PREFIX_PACKAGE_DATA_DIR@/properties; fi
 
 clean-local:
-	@rm -f system.tags
-
-system.tags:
-	cd $(top_builddir)/tagmanager && make libctags.a
-	cd $(top_builddir)/global-tags && make system.tags
-	mv $(top_builddir)/global-tags/system.tags .
