diff -urN anjuta-2.28.1.0/anjuta.desktop.in.in anjuta-2.28.1.0.new/anjuta.desktop.in.in
--- anjuta-2.28.1.0/anjuta.desktop.in.in	2009-09-26 12:05:29.000000000 +0200
+++ anjuta-2.28.1.0.new/anjuta.desktop.in.in	2009-12-26 16:41:27.219667561 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=GNOME;GTK;Development;
+Categories=GNOME;GTK;Development;IDE;
 StartupNotify=true
 MimeType=application/x-anjuta;
 X-GNOME-DocPath=anjuta/anjuta-manual.xml
