diff -urN anjuta-2.26.0.0/scripts/build-schemas.mk anjuta-2.26.0.0.new/scripts/build-schemas.mk
--- anjuta-2.26.0.0/scripts/build-schemas.mk	2009-03-04 15:56:06.000000000 +0100
+++ anjuta-2.26.0.0.new/scripts/build-schemas.mk	2009-03-18 23:06:47.000000000 +0100
@@ -2,6 +2,7 @@
 
 prefs_glade_schemasdir = @GCONF_SCHEMA_FILE_DIR@
 prefs_glade_schemas = $(prefs_glade_files:.glade=.schemas)
+prefs_glade_schemas_DATA = $(prefs_glade_schemas)
 
 %.schemas: %.glade
 	$(top_srcdir)/scripts/glade2schema.pl $(srcdir)/$(?) > $(@)
