diff -urN aaa/anjuta-1.2.0/anjuta.desktop.in.in anjuta-1.2.0/anjuta.desktop.in.in
--- aaa/anjuta-1.2.0/anjuta.desktop.in.in	2003-09-09 08:53:07.000000000 +0200
+++ anjuta-1.2.0/anjuta.desktop.in.in	2003-12-12 16:12:53.000000000 +0100
@@ -4,8 +4,8 @@
 Exec=anjuta %F
 Icon=@PACKAGE_PIXMAPS_DIR@/anjuta_icon.png
 Terminal=false
-MultipleArgs=false
+X-MultipleArgs=false
 Type=Application
 Categories=Application;Development;
-Encodings=UTF-8
+Encoding=UTF-8
 StartupNotify=true
