--- configure.in.org	2007-01-14 19:48:06.228659750 +0100
+++ configure.in	2007-01-14 19:48:26.069899750 +0100
@@ -893,7 +893,7 @@
 	test $VALUE && APR_CONFIG=$VALUE
 	if test $APR_CONFIG ; then
 		AC_MSG_RESULT([found])
-		APR_CFLAGS="`$APR_CONFIG --cflags`"
+		APR_CFLAGS="`$APR_CONFIG --cppflags --cflags`"
 		APR_INCLUDE="`$APR_CONFIG --includes`"
 		APR_LIBS="`$APR_CONFIG --link-ld --libs`"
 	else
